src/components/Clock.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+
+type Props = {
+  name: number;
+};
+
+type State = {
+  timerId: NodeJS.Timer | undefined;
+  date: Date;
+};
+
+export class Clock extends React.Component<Props, State> {
+  state = {
+    date: new Date(),
+    timerId: undefined,
+  };
+
+  componentDidMount() {
+    this.setState({
+      timerId: setInterval(() => {
+        this.setState({ date: new Date() });
+
+        // eslint-disable-next-line
+        console.log(this.state.date.toLocaleTimeString());
+      }, 1000),
+    });
```

Comment:
hard to understand the code

```suggestion
    const timerId = setInterval(() => {
        this.setState({ date: new Date() });

        // eslint-disable-next-line
        console.log(this.state.date.toLocaleTimeString());
      }, 1000),
      
    this.setState({ timerId });
```